src/components/Topbar/Topbar.ts

```diff
@@ -32,23 +57,92 @@ export class CandyTopbar extends LitElement {
     }
   };
 
+  openDrawer() {
+    this.burgerButton.classList.add("active");
+    this.drawer.classList.remove("display-none");
+    this.drawer.classList.remove("topbar-drawer-close");
+    this.drawer.classList.add("topbar-drawer-active");
+  }
+
+  closeDrawer() {
+    this.burgerButton.classList.remove("active");
+    this.drawer.classList.remove("topbar-drawer-active");
+    this.drawer.classList.add("topbar-drawer-close");
+    setTimeout(() => {
+      this.drawer.classList.add("display-none");
+    }, 150);
+  }
+
+  emitValueChangeEvent = (openValue: boolean) => {
+    const eventOnChange = new CustomEvent("onchange", {
+      bubbles: false,
+      composed: true,
+      detail: { value: openValue },
+    });
+    this.dispatchEvent(eventOnChange);
+  };
+
+  handleBurgerClick = (openValue: boolean) => {
+    this.isopen = openValue;
+    this.emitValueChangeEvent(openValue);
+    if (openValue) {
+      this.openDrawer();
+    } else {
+      this.closeDrawer();
+    }
+  };
+
   render() {
-    return html`
-      <header class="topbar-container" part="topbar">
-        <div class="${this.mobile ? "logo-reduced" : "logo"}">
-          <a href="/">
-            ${this.mobile
-              ? html`<candy-logo-mark></candy-logo-mark>`
-              : html`<candy-logo-horizontal></candy-logo-horizontal>`}
-          </a>
+    const displaySlot = html`<slot></slot>`;
+
+    const displayBurger = html` <button
+      part="tobpar-burger"
+      id="topbar-burger"
+      @click=${() => this.handleBurgerClick(!this.isopen)}
```

Comment:
@click=${this.handleMenuState}

Reply:
directly change isopen value to trigger update,

Done

Reply:
ok